src/disks.py

```diff
@@ -0,0 +1,49 @@
+import dataclasses
+from typing import Optional
+
+import dbus
+from moulinette.utils.log import getActionLogger
+
+logger = getActionLogger("yunohost.storage")
+
+
+UDISK_DRIVE_PATH = "/org/freedesktop/UDisks2/drives/"
+UDISK_DRIVE_IFC = "org.freedesktop.UDisks2.Drive"
+
+
+@dataclasses.dataclass
+class DiskInfos:
+    name: str
+    model: str
+    serial: str
+    size: int
+    type: str
+    rpm: Optional[int] = None
+
+
+def infos():
+    result = {}
+
+    bus = dbus.SystemBus()
+    manager = bus.get_object("org.freedesktop.UDisks2", "/org/freedesktop/UDisks2")
+
+    for k, v in manager.get_dbus_method(
+        "GetManagedObjects", "org.freedesktop.DBus.ObjectManager"
+    )().items():
```

Comment:
Does someone knows a more efficient way to interact with dbus than just list *all the objects managed by Udisks2* here? Or is dbus definitely a crap?